dhis-2/dhis-support/dhis-support-commons/src/main/java/org/hisp/dhis/commons/collection/CollectionUtils.java

```diff
@@ -43,6 +45,42 @@
 {
     public static final String[] STRING_ARR = new String[0];
 
+    /**
+     * Performs a flat mapping of the given collection using the given mapping
+     * function.
+     *
+     * @param <T>
+     * @param <U>
+     * @param collection the collection of objects to map.
+     * @param mapper the mapping function.
+     * @return a set of mapped objects.
+     */
+    public static <T, U> Set<U> flatMapToSet( Collection<T> collection,
```

Comment:
Just a suggestion: for mapping it often reads more intuitive if we call the generics `A` and `B` instead of `T` and `U`.

Reply:
Alright, updated it.